src/pyscipopt/expr.pxi

```diff
@@ -1,3 +1,47 @@
+# In this file we implemenet the handling of expressions
+# We have two types of expressions: Expr and GenExpr.
+# The Expr can only handle polynomial expressions.
+# In addition, one can recover easily information from them.
+# A polynomial is a dictionary between `terms` and coefficients.
+# A `term` is a tuple of variables
+# For examples, 2*x*x*y*z - 1.3 x*y*y + 1 is stored as a
+# {Term(x,x,y,z) : 2, Term(x,y,y) : -1.3, Term() : 1}
+# Addition of common terms and expansion of exponents occur automatically.
+# Given the way `Expr`s are stored, it is easy to access the terms: e.g.
+# expr = 2*x*x*y*z - 1.3 x*y*y + 1
+# expr[Term(x,x,y,z)] returns 1.3
+# expr[Term(x)] returns 0.0
+#
+# On the other hand, when dealing with expressions more general than polynomials,
+# that is, absolute values, exp, log, sqrt or any general exponent, we use GenExpr.
+# GenExpr stores expression trees in a rudimentary way.
+# Basically, it stores the operator and the list of children.
+# We have different types of general expressions that in addition
+# to the operation and list of children stores
+# SumExpr: coefficients and constant
+# ProdExpr: constant
```

Comment:
Nice documentation!

But what is the meaning of the constant for `ProdExpr`? Is it `contant * child1 * child2 * ...` ?

Reply:
yes